### ISSUE TITLE
Atom selection slightly off when in ball and stick representation mode

### ISSUE THREAD

**Issue by ItsRaminTime** [biasmv/pv#172]
Hi!

When I'm trying to select an atom and highlight it in the ball and stick representation, following the selection guide for the cartoon representation, the selection always seems to be off by one atom. 
Any explanation or help for this?
(As an example: in the image below I clicked on the atom to the left of the red highlight but the highlight always happens one atom to the right)

Thanks!!

<img width="249" alt="screen shot 2017-02-08 at 2 27 40 pm" src="https://cloud.githubusercontent.com/assets/13488908/22760324/de3f1758-ee0a-11e6-8da4-e5578b4e9e84.png">

**Comment by biasmv** [comment]
that's not supposed to happen. This could be a bug in the selection code. What browser are you using?  Could you paste the code for a sample to reproduce this issue?

**Comment by ItsRaminTime** [comment]
Yes of course, pasting it below. I'm using Google Chrome but observed this in firefox as well. 

viewer.on('click', function(picked, ev) {
  /*var atom = picked.target();
  if (picked === null || picked.target() === null) {
    return;
  }
  // don't to anything if the clicked structure does not have an atom.
  if (picked.node().structure === undefined) {
    return;
  }
  // when the shift key is pressed, extend the selection, otherwise
  // only select the clicke atom.
  var extendSelection = ev.shiftKey;
  var sel;
  if (extendSelection) {
    var sel = picked.node().selection();
  } else {
    var sel = picked.node().structure().createEmptyView();
  }
  // in case atom was not part of the view, we have to add it, because
  // it wasn't selected before. Otherwise removeAtom took care of it
  // and we don't have to do anything.
  if (!sel.removeAtom(picked.target(), true)) {
    sel.addAtom(picked.target());
  }
  picked.node().setSelection(sel);
  viewer.requestRedraw();
 */    
    var rect = viewer.boundingClientRect();
    var picked = viewer.pick({ x : ev.clientX - rect.left,
                               y : ev.clientY - rect.top });
    if (prevPicked !== null && picked !== null &&
        picked.target() === prevPicked.atom) {
      return;
    }
    if (prevPicked !== null) {
      // reset color of previously picked atom.
      setColorForAtom(prevPicked.node, prevPicked.atom, prevPicked.color);
    }
    if (picked !== null) {
      var atom = picked.target();
      document.getElementById('picked-atom-name').innerHTML = atom.qualifiedName();
      // get RGBA color and store in the color array, so we know what it was
      // before changing it to the highlight color.
      var color = [0,0,0,0];
      picked.node().getColorForAtom(atom, color);
      prevPicked = { atom : atom, color : color, node : picked.node() };

      setColorForAtom(picked.node(), atom, 'red');
    } else {
      document.getElementById('picked-atom-name').innerHTML = '&nbsp;';
      prevPicked = null;
    }
    viewer.requestRedraw();
    
  document.getElementById('picked-atom-name').innerHTML = atom.qualifiedName();
  var atomNum = atom.qualifiedName().match(/\d+/)[0];
  toggleResi(atomNum);
});

I tried both the commented out and the not commented out parts and they both led to the same error.

**Comment by ItsRaminTime** [comment]
`viewer.on('click', function(picked, ev) {
    var rect = viewer.boundingClientRect();
    var picked = viewer.pick({ x : ev.clientX - rect.left,
                               y : ev.clientY - rect.top });
    if (prevPicked !== null && picked !== null &&
        picked.target() === prevPicked.atom) {
      return;
    }
    if (prevPicked !== null) {
      // reset color of previously picked atom.
      setColorForAtom(prevPicked.node, prevPicked.atom, prevPicked.color);
    }
    if (picked !== null) {
      var atom = picked.target();
      document.getElementById('picked-atom-name').innerHTML = atom.qualifiedName();
      // get RGBA color and store in the color array, so we know what it was
      // before changing it to the highlight color.
      var color = [0,0,0,0];
      picked.node().getColorForAtom(atom, color);
      prevPicked = { atom : atom, color : color, node : picked.node() };

      setColorForAtom(picked.node(), atom, 'red');
    } else {
      document.getElementById('picked-atom-name').innerHTML = '&nbsp;';
      prevPicked = null;
    }
    viewer.requestRedraw();
    
  document.getElementById('picked-atom-name').innerHTML = atom.qualifiedName();
  var atomNum = atom.qualifiedName().match(/\d+/)[0];
  toggleResi(atomNum);
});`
